src/legacy/core_plugins/vega/public/vega_visualization.js

```diff
@@ -17,117 +17,121 @@
  * under the License.
  */
 
-import { i18n }  from '@kbn/i18n';
+import { i18n } from '@kbn/i18n';
 import { toastNotifications } from 'ui/notify';
 import { VegaView } from './vega_view/vega_view';
 import { VegaMapView } from './vega_view/vega_map_view';
-import { SavedObjectsClientProvider, findObjectByTitle } from 'ui/saved_objects';
-
-// $rootScope is for the removeFilter() workaround, see vega_view/vega_base_view.js
-export function VegaVisualizationProvider(Private, vegaConfig, serviceSettings, $rootScope) {
+import { findObjectByTitle } from 'ui/saved_objects';
+
+// Temporary solution
+import chrome from 'ui/chrome';
+
+export class VegaVisualization {
+  constructor(el, vis) {
+    this._el = el;
+    this._vis = vis;
+  }
+
+  /**
+   * Find index pattern by its title, of if not given, gets default
+   * @param {string} [index]
+   * @returns {Promise<string>} index id
+   */
+  async findIndex(index) {
+    let idxObj;
+    if (index) {
+      idxObj = await findObjectByTitle(this.savedObjectsClient, 'index-pattern', index);
+      if (!idxObj) {
+        throw new Error(i18n.translate('vega.visualization.indexNotFoundErrorMessage', {
+          defaultMessage: 'Index {index} not found',
+          values: { index: `"${index}"` },
+        }));
+      }
+    } else {
+      idxObj = await this._vis.API.indexPatterns.getDefault();
+      if (!idxObj) {
+        throw new Error(i18n.translate('vega.visualization.unableToFindDefaultIndexErrorMessage', {
+          defaultMessage: 'Unable to find default index',
+        }));
+      }
+    }
+    return idxObj.id;
+  }
+
+  /**
+   *
+   * @param {VegaParser} visData
+   * @param {*} status
+   * @returns {Promise<void>}
+   */
+  async render(visData, visParams, status) {
+
+    // todo: work in progress.
+    const $injector = await chrome.dangerouslyGetActiveInjector();
+    this.savedObjectsClient = chrome.getSavedObjectsClient();
+    this.vegaConfig = {}; // needs somehow to get  enableExternalUrls from \vega\index.ts;
```

Comment:
How I can get the  enableExternalUrls  value declared \vega\index.ts 

![image](https://user-images.githubusercontent.com/20072247/60445340-b375a500-9c27-11e9-9d09-987f123d199b.png)

Reply:
It looks possible that the replacement for this in the new platform will be `core.injectedMetadata.getInjectedVar`; will let you know once I find out for sure.